t/mojo/json_pointer.t

```diff
@@ -3,94 +3,105 @@ use Mojo::Base -strict;
-ok $pointer->contains('/foo'), 'contains "/foo"';
-ok !$pointer->contains('foo'),  '"foo" is invalid';
-ok !$pointer->contains('/bar'), 'does not contains "/bar"';
-ok $pointer->new({foo => {bar => undef}})->contains('/foo/bar'), 'contains "/foo/bar"';
+subtest '"contains" (hash)' => sub {
+  my $pointer = Mojo::JSON::Pointer->new({foo => 23, '' => 24});
+  ok $pointer->contains(''),     'contains ""';
+  ok $pointer->contains('/'),    'contains "/"';
+  ok $pointer->contains('/foo'), 'contains "/foo"';
+  ok !$pointer->contains('foo'),  '"foo" is invalid';
+  ok !$pointer->contains('/bar'), 'does not contains "/bar"';
+  ok $pointer->new({foo => {bar => undef}})->contains('/foo/bar'), 'contains "/foo/bar"';
+};
 
-# "contains" (string)
-$pointer = Mojo::JSON::Pointer->new('works');
-ok $pointer->contains(''), 'contains ""';
-ok !$pointer->contains('/'),    'does not contain "/"';
-ok !$pointer->contains('/foo'), 'does not contain "/foo"';
-ok $pointer->new('0')->contains(''), 'contains ""';
+subtest '"contains" (string)' => sub {
+  my $pointer = Mojo::JSON::Pointer->new('works');
+  ok $pointer->contains(''), 'contains ""';
+  ok !$pointer->contains('/'),    'does not contain "/"';
+  ok !$pointer->contains('/foo'), 'does not contain "/foo"';
+  ok $pointer->new('0')->contains(''), 'contains ""';
+};
 
-# "contains" (mixed)
-$pointer = Mojo::JSON::Pointer->new({foo => [0, 1, 2]});
-ok $pointer->contains(''),       'contains ""';
-ok $pointer->contains('/foo/0'), 'contains "/foo/0"';
-ok !$pointer->contains('/foo/9'),   'does not contain "/foo/9"';
-ok !$pointer->contains('/foo/bar'), 'does not contain "/foo/bar"';
-ok !$pointer->contains('/0'),       'does not contain "/0"';
-ok !$pointer->contains('/'),        'does not contain "/"';
+subtest '"contains" (mixed)' => sub {
+  my $pointer = Mojo::JSON::Pointer->new({foo => [0, 1, 2]});
+  ok $pointer->contains(''),       'contains ""';
+  ok $pointer->contains('/foo/0'), 'contains "/foo/0"';
+  ok !$pointer->contains('/foo/9'),   'does not contain "/foo/9"';
+  ok !$pointer->contains('/foo/bar'), 'does not contain "/foo/bar"';
+  ok !$pointer->contains('/0'),       'does not contain "/0"';
+  ok !$pointer->contains('/'),        'does not contain "/"';
+};
 
-# "get" (hash)
-$pointer = Mojo::JSON::Pointer->new({foo => 'bar', '' => 'baz'});
-is_deeply $pointer->get(''), {foo => 'bar', '' => 'baz'}, '"" is "{foo => "bar", "" => "baz"}"';
-is $pointer->get('/'),    'baz', '"/" is "baz"';
-is $pointer->get('/foo'), 'bar', '"/foo" is "bar"';
-is $pointer->get('foo'),  undef, '"foo" is invalid';
-is $pointer->new({foo => {bar => 42}})->get('/foo/bar'), 42, '"/foo/bar" is "42"';
-is_deeply $pointer->new({foo => {23 => {baz => 0}}})->get('/foo/23'), {baz => 0}, '"/foo/23" is "{baz => 0}"';
-is $pointer->new({foo => {'' => 42}})->get('/foo/'), 42, '"/foo/" is "42"';
-is $pointer->new({foo => {'' => {'' => 42}}})->get('/foo//'), 42, '"/foo//" is "42"';
+my $pointer;
```

Comment:
Why is this before the block?

Reply:
That's what I was trying express in my prior comment, namely that the "get" tests appear to share the $pointer variable across different subsections.  For example, note how the first line in the '"get" (mixed)' section runs is_deeply.  Thus, I pulled the variable out to reflect that sharing.  Is there a better way to handle this?

Reply:
It would be better to have each subtest set up $pointer again as it needs. There isn't anything that actually needs to be shared.

Reply:
Sounds good, I will have another go at this then, thanks!